src/globals/scss/core/_typography.scss

```diff
@@ -1,435 +1,173 @@
-    @include govuk-copy-14;
-  }
+  .govuk-body,
+  .govuk-body-m {
+    @include govuk-font-regular-19;
 
-  .govuk-u-copy-16 {
-    @include govuk-copy-16;
-  }
+    margin-top: 0;
+    margin-bottom: $govuk-spacing-scale-3;
 
-  .govuk-u-copy-19 {
-    @include govuk-copy-19;
+    @include mq($from: tablet) {
+      margin-bottom: $govuk-spacing-scale-4;
+    }
   }
 
-  // Copy headings with extra margins from GOV.UK elements
-  .heading-xlarge {
-    @include govuk-bold-48;
+  .govuk-body-s {
+    @include govuk-font-regular-16;
 
-    margin-top: em($govuk-spacing-scale-3, 32);
-    margin-bottom: em($govuk-spacing-scale-5, 32);
+    margin-top: 0;
+    margin-bottom: $govuk-spacing-scale-3;
 
     @include mq($from: tablet) {
-      margin-top: em($govuk-spacing-scale-5, 48);
-      margin-bottom: em($govuk-spacing-scale-6, 48);
-    }
-
-    .heading-secondary {
-      @include govuk-heading-27;
-
-      display: block;
-      color: $govuk-secondary-text-colour;
+      margin-bottom: $govuk-spacing-scale-4;
     }
-
   }
 
-  .heading-large {
-    @include govuk-bold-36;
+  .govuk-body-xs {
+    @include govuk-font-regular-14;
 
-    margin-top: em(25, 24);
-    margin-bottom: em($govuk-spacing-scale-2, 24);
+    margin-top: 0;
+    margin-bottom: $govuk-spacing-scale-3;
 
     @include mq($from: tablet) {
-      margin-top: em(45, 36);
-      margin-bottom: em($govuk-spacing-scale-4, 36);
+      margin-bottom: $govuk-spacing-scale-4;
     }
+  }
 
-    .heading-secondary {
-      @include govuk-heading-24;
+  // Contextual adjustments
 
-      display: block;
-      color: $govuk-secondary-text-colour;
-    }
+  // Add a negative top margin to headings that appear directly after other
+  // headings
 
+  .govuk-heading-l + .govuk-heading-m {
+    margin-top: -$govuk-spacing-scale-1;
   }
 
-  .heading-medium {
-    @include govuk-bold-24;
+  .govuk-heading-m + .govuk-heading-s {
+    margin-top: -$govuk-spacing-scale-2;
+  }
+
+  // Add top padding to headings that appear directly after paragraphs
 
-    margin-top: em(25, 20);
-    margin-bottom: em($govuk-spacing-scale-2, 20);
+  .govuk-body-l  + .govuk-heading-l {
+    padding-top: $govuk-spacing-scale-1;
 
     @include mq($from: tablet) {
-      margin-top: em(45, 24);
-      margin-bottom: em($govuk-spacing-scale-4, 24);
+      padding-top: $govuk-spacing-scale-2;
     }
-
   }
 
-  .heading-small {
-    @include govuk-bold-19;
-
-    margin-top: em($govuk-spacing-scale-2, 16);
-    margin-bottom: em($govuk-spacing-scale-1, 16);
+  .govuk-body-m  + .govuk-heading-l,
+  .govuk-body-s  + .govuk-heading-l {
+    padding-top: $govuk-spacing-scale-3;
 
     @include mq($from: tablet) {
-      margin-top: em($govuk-spacing-scale-4, 19);
+      padding-top: $govuk-spacing-scale-4;
     }
   }
 
-  .body-text {
-    @include govuk-core-19;
-    margin-top: em($govuk-spacing-scale-1, 16);
-    margin-bottom: em($govuk-spacing-scale-4, 16);
+  .govuk-body-m + .govuk-heading-m,
```

Comment:
Apologies I missed this... this should also include `govuk-body-m` + `govuk-heading-s` and `govuk-body-s` + `govuk-heading-s`

Reply:
Pushed a fix